subsys/bluetooth/host/Kconfig

```diff
@@ -320,6 +320,45 @@ config BT_PRIVACY
 	  Enable local Privacy Feature support. This makes it possible
 	  to use Resolvable Private Addresses (RPAs).
 
+config BT_PRIVACY_RANDOMIZE_IR
+	bool "Randomize identity root for fallback identities"
+	depends on BT_PRIVACY
+	select BT_SETTINGS
+	help
+	  Enabling this option will cause the Host to ignore controller-provided
+	  identity roots (IR). The Host will instead use bt_rand to generate
+	  identity resolving keys (IRK) and store them in non-volatile memory.
+	  This may come with a performance penalty.
+
+	  This option increases privacy as explained in the following text.
+
+	  The IR determines the IRK of the identity. The IRK is used to both
+	  generate and resolve (recognize) the private address of an identity.
+	  The IRK is distributed to bonded devices.
+
+	  An attacker that has stolen or once bonded and retained the IRK can
+	  forever resolve addresses from that IRK, even if that bond has been
+	  deleted locally.
+
+	  If the application has not setup an identity before calling
+	  settings_load() or settings_load_subtree("bt") before bt_enable(),
+	  the Host will fallback to automatically setup the default identity
+	  (BT_ID_DEFAULT).
+
+	  If the controller has a public address (HCI_Read_BD_ADDR) that becomes
+	  the address of the fallback identity. In that case, this config
+	  determines if the Host should try to use the general identity root
+	  from the controller (Zephyr HCI Read_Key_Hierarchy_Roots) for that
+	  identity.
```

Comment:
Could we change `this config determines` to `setting this config to false will make the Host try to use ...` This way it's clearer what value does what. same comment for the next paragraph.